src/vmax.rs

```diff
@@ -1188,3 +1192,191 @@ impl ChildPoint for PhysicalCapacity {
+    //pub rdf_group_id: Option<Vec<u64>>,
+    pub snapvx_source: Option<bool>,
+    pub snapvx_target: Option<bool>,
+    pub cu_image_base_address: Option<String>,
+    pub has_effective_wwn: Option<bool>,
+    pub effective_wwn: Option<String>,
+    pub encapsulated_wwn: Option<String>,
+}
+
+#[test]
+fn test_get_vmax_json_volume() {
+    use std::fs::File;
+    use std::io::Read;
+
+    let mut f = File::open("tests/vmax/slo_volume.json").unwrap();
+    let mut buff = String::new();
+    f.read_to_string(&mut buff).unwrap();
+
+    let i: Volume = serde_json::from_str(&buff).unwrap();
+    println!("result: {:#?}", i);
+    println!("point: {:#?}", i.into_point(Some("vmax_slo_volume"), false));
+}
+
+#[test]
+/// Test retrieving one page of volumes from json
+fn test_get_vmax_json_volumes() {
+    use std::fs::File;
+    use std::io::Read;
+
+    let mut f = File::open("tests/vmax/slo_volumes.json").unwrap();
+    let mut buff = String::new();
+    f.read_to_string(&mut buff).unwrap();
+
+    let data: Value = serde_json::from_str(&buff).unwrap();
+
+    // grab the list from what was returned before calling the iterator
+    let all_volume_ids = match data["resultList"]["result"].as_array() {
+        Some(v) => v
+            .iter()
+            .map(|val| {
+                val.as_object()
+                    .unwrap()
+                    .values()
+                    .map(|v_i| v_i.as_str().unwrap().to_string())
+                    .collect::<String>()
+            })
+            .collect(),
+        None => vec![],
+    };
+
+    println!("Volume IDs: {:?}", all_volume_ids);
+}
+
+/// Returns a list of volume IDs for this array
+// TODO: combine this with the other getters and generalize along with other metrics
+pub fn get_all_slo_volumes(
+    client: &reqwest::Client,
+    config: &VmaxConfig,
+    symmetrixid: &str,
+) -> MetricsResult<Vec<String>> {
+    let data: Value = super::get(
+        client,
+        &format!(
+            "https://{}/univmax/restapi/90/sloprovisioning/symmetrix/{}/volume",
+            config.endpoint, symmetrixid
+        ),
+        &config.user,
+        Some(&config.password),
+    )?;
+
+    let vol_count = match data["count"].as_u64() {
+        Some(count) => count,
+        None => 0,
+    };
+    let iterator_id = match data["id"].as_str() {
+        Some(id) => id,
+        None => "",
+    };
+    let max_count_per_page = match data["max_page_size"].as_u64() {
+        Some(count) => count,
+        None => 0,
+    };
+
+    if vol_count == 0 || max_count_per_page == 0 || iterator_id.is_empty() {
+        return Ok(vec![]);
+    }
+    let mut num_iterations = vol_count / max_count_per_page;
+
+    // grab the list from what was returned before calling the iterator
+    let mut all_volume_ids = match data["resultList"]["result"].as_array() {
+        Some(v) => v
+            .iter()
+            .map(|val| {
+                val.as_object()
+                    .unwrap()
```

Comment:
I know we know it's an object here based on the tests but I'm wondering if it'd be better to use `expect("vmax returned an invalid object")` or something like that so we know what happened.